File: alumni-intelligence-service/src/main/java/edu/mbstu/alumni/model/AlumniProfile.java
package edu.mbstu.alumni.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "alumni_profiles")
public class AlumniProfile {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "user_id")
    private Long userId;
    
    @Column(name = "department_id")
    private Long departmentId;
    
    @Column(name = "student_id")
    private String studentId;
    
    @Column(name = "batch_year")
    private Integer batchYear;
    
    @Column(name = "graduation_year")
    private Integer graduationYear;
    
    private String degree;
    private String major;
    
    @Column(name = "profile_photo")
    private String profilePhoto;
    
    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;
    
    private String phone;
    private String address;
    private String city;
    private String country;
    
    @Column(name = "current_company")
    private String currentCompany;
    
    @Column(name = "current_position")
    private String currentPosition;
    
    private String industry;
    
    @Column(name = "linkedin_url")
    private String linkedinUrl;
    
    @Column(name = "facebook_url")
    private String facebookUrl;
    
    @Column(name = "twitter_url")
    private String twitterUrl;
    
    @Column(name = "website_url")
    private String websiteUrl;
    
    @Column(columnDefinition = "TEXT")
    private String bio;
    
    @Column(columnDefinition = "JSON")
    private String achievements;
    
    @Column(columnDefinition = "JSON")
    private String publications;
    
    @Column(name = "is_profile_public")
    private Boolean isProfilePublic;
    
    @Column(name = "is_verified")
    private Boolean isVerified;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", insertable = false, updatable = false)
    private Department department;
}

File: app/Models/AlumniProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class AlumniProfile extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'department_id',
        'student_id',
        'batch_year',
        'graduation_year',
        'degree',
        'major',
        'profile_photo',
        'date_of_birth',
        'phone',
        'address',
        'city',
        'country',
        'current_company',
        'current_position',
        'industry',
        'linkedin_url',
        'facebook_url',
        'twitter_url',
        'website_url',
        'bio',
        'achievements',
        'publications',
        'is_profile_public',
        'is_verified',
    ];

    protected function casts(): array
    {
        return [
            'date_of_birth' => 'date',
            'is_profile_public' => 'boolean',
            'is_verified' => 'boolean',
            'achievements' => 'array',
            'publications' => 'array',
        ];
    }

    // Relationships
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function department()
    {
        return $this->belongsTo(Department::class);
    }

    // Helper methods
    public function getFullNameAttribute(): string
    {
        return $this->user->name;
    }

    public function getProfilePhotoUrlAttribute(): string
    {
        return $this->profile_photo 
            ? asset('storage/' . $this->profile_photo)
            : asset('images/default-avatar.png');
    }

    public function getBatchMates()
    {
        return self::where('department_id', $this->department_id)
            ->where('batch_year', $this->batch_year)
            ->where('id', '!=', $this->id)
            ->get();
    }
}
